src/libsync/utility_win.cpp

```diff
@@ -15,21 +15,56 @@
 #include <shlobj.h>
 #include <winbase.h>
 #include <windows.h>
+#include <shlguid.h>
+#include <string>
+#include <QLibrary>
 
 static const char runPathC[] = "HKEY_CURRENT_USER\\Software\\Microsoft\\Windows\\CurrentVersion\\Run";
 
+typedef HRESULT (WINAPI *SHGetKnownFolderPathFun)(
+    const GUID &rfid, 
+    DWORD dwFlags, 
+    HANDLE hToken, 
+    PWSTR *ppszPath
+);
+     
 static void setupFavLink_private(const QString &folder)
 {
     // Windows Explorer: Place under "Favorites" (Links)
-    wchar_t path[MAX_PATH];
-    SHGetSpecialFolderPath(0, path, CSIDL_PROFILE, FALSE);
-    QString profile =  QDir::fromNativeSeparators(QString::fromWCharArray(path));
+    
+    static SHGetKnownFolderPathFun SHGetKnownFolderPathPtr = NULL;
+    QString linkName; 
     QDir folderDir(QDir::fromNativeSeparators(folder));
-    QString linkName = profile+QLatin1String("/Links/") + folderDir.dirName() + QLatin1String(".lnk");
+    if (!SHGetKnownFolderPathPtr)
+    {
+      QLibrary kernel32Lib("shell32.dll");
+      if(kernel32Lib.load())
+      {
+        SHGetKnownFolderPathPtr = (SHGetKnownFolderPathFun) kernel32Lib.resolve("SHGetKnownFolderPath");
```

Comment:
Last time i tried this i had errors with direct lookup, but maybe because was testing also XP clients. If you say that Windows 7 is the minimum supported, i can try rewriting it